libsolidity/analysis/ReferencesResolver.cpp

```diff
@@ -184,6 +185,35 @@ void ReferencesResolver::endVisit(IdentifierPath const& _path)
 	_path.annotation().pathDeclarations = std::move(declarations);
 }
 
+bool ReferencesResolver::visit(UsingForDirective const& _usingFor)
+{
+	for (ASTPointer<IdentifierPath> const& path: _usingFor.functionsOrLibrary())
+	{
+		// _includeInvisibles is enabled here because external library functions are marked invisible.
+		// As unintended side-effects other invisible names (eg.: super, this) may be returned as well.
+		// DeclarationTypeChecker should detect and report such situations.
+		vector<Declaration const*> declarations = m_resolver.pathFromCurrentScopeWithAllDeclarations(path->path(), true /* _includeInvisibles */);
+		if (declarations.empty())
+		{
+			string libraryOrFunctionName = _usingFor.usesBraces() ? "Function" : "Library";
+			m_errorReporter.fatalDeclarationError(
+				9589_error,
+				path->location(),
+				libraryOrFunctionName + " name expected, but the identifier was not found or is not unique."
```

Comment:
Something is bugging me a little about this wording. For example it seems to imply that what you used is not a function but you'll get this message when you use an overloaded function. It does say "or is not unique" so it's not incorrect, but still I wonder if we could word it better. And for libraries the name can't really be non-unique - not 100% sure but I think that conflicting names would be a fatal error and an earlier analysis stage.

What do you think about doing this instead?
- `Identifier is not a function name or not unique.`
- `Identifier is not a library name.`

Reply:
I think it is a good idea. Done.

Reply:
The